tripal/src/TripalImporter/Attribute/TripalImporter.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Drupal\tripal\TripalImporter\Attribute;
+
+use Drupal\Component\Plugin\Attribute\Plugin;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+
+/**
+ * Defines a Tripal Importer attribute object.
+ *
+ * Plugin Namespace: Drupal\tripal\TripalImporter
+ *
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class TripalImporter extends Plugin {
+
+  /**
+   * Constructs a Tripal Importer attribute.
+   *
+   * @param string $id
+   *   The plugin ID.
+   * @param \Drupal\Core\StringTranslation\TranslatableMarkup|null $label
+   *   The label of the importer plugin.
+   * @param \Drupal\Core\StringTranslation\TranslatableMarkup|null $description
+   *   A brief description for this loader.
+   *   This description will be presented to the site user.
+   * @param array|null $file_types
+   *   An array containing the extensions of allowed file types.
+   * @param \Drupal\Core\StringTranslation\TranslatableMarkup|null $upload_description
+   *   Provides information to the user about the file upload.
+   *   Typically this may include a description of the file types allowed.
+   * @param \Drupal\Core\StringTranslation\TranslatableMarkup|null $upload_title
+   *   The title that should appear above the upload button.
+   * @param bool|null $use_analysis
+   *   If the loader should require an analysis record.
+   *   To maintain provenance we should always indicate where the data we are
+   *   uploading comes from. The method that Tripal attempts to use for this
+   *   by associating upload files with an analysis record. The analysis
+   *   record provides the details for how the file was created or obtained.
+   *   Set this to FALSE if the loader should not require an analysis when
+   *   loading. if $use_analysis is set to true then the form values will
+   *   have an 'analysis_id' key in the $form_state array on submitted forms.
+   * @param bool|null $require_analysis
+   *   If the $use_analysis value is set above then this value indicates if the
+   *   analysis should be required.
+   * @param bool|null $use_button
+   *   Indicates whether the base importer should add a submit button or not.
+   *   This should only be used in situations were you need multiple buttons
+   *   or control over the submit process (e.g. multi-page forms).
+   * @param bool|null $submit_enabled
+   *   Indicated whether the base importer added submit button should be disabled
+   *   when the form is first loaded (i.e when the user clicks the link for the
+   *   importer). The form can then be programatically enabled via AJAX once
+   *   certain criteria is set by setting the form state storage.
+   *   Example of programatically enabling the button via the form state
+   *   on an importer where this annotation is set to TRUE.
```

Comment:
Coding standards and misnamed param

```suggestion
   * @param bool|null $submit_disabled
   *   Used to disabled the base importer added submit button when the form
   *   is first loaded (i.e when the user clicks the link for the importer). 
   *   The form can then be programatically enabled via AJAX by setting the 
   *   form state storage as shown in the following example.
```